File: DataStructure/Queue.py
class Queue:
    front = 0
    rear = 0
    size = 0
    queue = []

    def __init__(self, size):
        self.queue = [-1 for i in range(0,size)]
        self.size = size
        self.front = 0
        self.rear = 0
    
    def is_empty(self):
        if self.rear is self.front:
            return True
        return False
    
    def is_full(self):
        if(self.rear + 1) % size is self.front:
            return True
        return False

    def enqueue(self, item):
        if self.is_full():
            return
        self.rear = (self.rear + 1) % self.size
        self.queue[self.rear] = item

    def dequeue(self):
        if self.is_empty():
            return
        self.front = (self.front + 1) % self.size
        self.queue[self.front] = -1
        

File: DataStructure/Queue.java
public class Queue{
    private static class Node {
        private T data;
        private Node next;

        public Queue(T data){
            this.data = data;
        }
    }

    private Node first;
    private Node last;

    public void add(T item){
        Node t = new Node(item);

        if(last != null) last.next = t;
        last = t;
        if(isEmpty()) first = last;
    }

    public T remove(){
        if(isEmpty()) throw new NoSuchElementException();
        T data = first.data;
        first = first.next;

        if(isEmpty()) last = null;
        return data;
    }

    public T peek(){
        if(isEmpty()) throw new NoSuchElementException();
        return first.data;
    }

    public boolean isEmpty(){
        return first == null;
    }
}